mlflow/pyfunc/utils/data_validation.py

```diff
@@ -110,12 +111,15 @@ def _get_func_info_if_type_hint_supported(func) -> Optional[FuncInfo]:
         try:
             _infer_schema_from_list_type_hint(type_hint)
         except Exception as e:
-            warnings.warn(
-                "Type hint used in the model's predict function is not supported "
-                f"for MLflow's schema validation. {e}. "
-                "To enable validation for the input data, specify the input example "
-                "or model signature when logging the model.",
+            color_warning(
+                message="Type hint used in the model's predict function is not supported "
+                f"for MLflow's schema validation. {e} "
+                "Remove the type hint to disable this warning. "
+                "To enable validation for the input data, specify input example "
+                "or model signature when logging the model. ",
```

Comment:
```suggestion
                "To enable input validation for the specified type, provide input example "
                "or model signature when logging the model. ",
```

nit: I don't think we should recommend removing type hints? (in terms of users' code maintainability)

Reply:
Emm but if they use type hints that we don't support, they will see this warning and might feel annoying 🤔